### ISSUE TITLE
Weird unit handling with multiplication and division

### ISSUE THREAD

**Issue by damskii9992** [scipp/scipp#3500]
The way that units are currently handled in Scipp in multiplication and division is very confusing and undesirable. 
Minimal example:
```python
import scipp as sc
test = sc.scalar(1.0, unit="m")
test2 = sc.scalar(1.0, unit="dm")
test * test2
```
The output is then printed with the unit: 0.1m^2
This "unit" is extremely confusing, as it is very unclear if this means (0.1m)^2 or 0.1*m^2.
I would suggest that the 0.1 is multiplied to the value, so that the output is 0.1 m^2 or 10 dm^2 to avoid this confusion.

**Comment by SimonHeybrock** [comment]
> I would suggest that the 0.1 is multiplied to the value, so that the output is 0.1 m^2 or 10 dm^2 to avoid this confusion.

In general this is a very complicated thing to do: How do you decide when to move scale factors into the values? I don't think there is any rule on when that should be done that every user would agree with.

There are cases such as pure scale factors (as discussed also in a comment #3456) where it is kind of obvious, but in all other cases I am not sure how you would define a general logic, and one might need to rely on some sorts of allow-lists and deny-lists, which would be very complicated.

**Comment by damskii9992** [comment]
Sure, formatting it as `dm * m` would be a perfectly sensible solution. As far as I recall, this is even how it is done in mathematical software such as Mathematica and Maple.
The unit could then be "simplified" to `m^2 ` by a simplify method or similar where the prefactor is absorbed in the value.

What I am struggling with in the current implementation, is the choice of having numerical prefactors in units. I have never seen that done before and I don't know how to interpret it, and I fear many others would be in a similar situation. For now though, in our usage of Scipp in EasyScience, we will be converting the 2nd operand to avoid this issue and align with other libraries such as Pint.

**Comment by SimonHeybrock** [comment]
> What I am struggling with in the current implementation, is the choice of having numerical prefactors in units.

Numerical prefactors is how SI prefixes (and other scale factors, such as for eV, feet, Angstrom, ...) are implemented in the underlying library. Unless you have a better idea and are willing to implement and maintain your own library, that is how it is.

It seems though that you are really more worried about formatting? Can you clarify if this is the case?

**Comment by SimonHeybrock** [comment]
Regarding Pint, the big difference is that Pint does no automatic cancellation of factors (base dims) in units, whereas Scipp does. That is, you will never be able to replace one with the other, without further code changes.

**Comment by damskii9992** [comment]
Oh, yes. This is purely a formatting issue. 
I am a little confused though, did you think I was against representing SI prefixes as numerical values internally?
My only concern is how the unit is presented to the user, and surely this can be changed in Scipp without changing the underlying unit library?

**Comment by SimonHeybrock** [comment]
Scipp uses the underlying library (https://github.com/LLNL/units) to format units. If there is a reasonable change suggestion then we would try to contribute it upstream.

**Comment by henrikjacobsenfys** [comment]
> In general this is a very complicated thing to do: How do you decide when to move scale factors into the values?

I would say always. 

I don't think it ever makes sense to have a unit of e.g. 0.1 m. 1 dm makes sense as a unit, but once a conversion from dm to 0.1 m has happened, that number should always be moved to the value. 

Just to make the example above even more clear:
I would much rather have (1) (m^2) than (10) (0.1 m^2). I don't mind (10) (dm*m).

**Comment by jl-wynen** [comment]
> I would say it's simple: you always do it.

This would mean that units are always in SI base units. E.g., `1 * sc.Unit('Å')` would be formatted as `"1e-10 m"`.

To explain the underlying mechanism a bit more, consider this:
```python
>>> import scipp as sc
>>> sc.Unit('Å').to_dict()
{'__version__': 2, 'multiplier': 1e-10, 'powers': {'m': 1}}
```
`to_dict()` returns the data contained in a unit, spelled out as a dict. So it has the base power and a multiplier. When formatting a unit or variable, Scipp and LLNL/units have to figure out how best to show the unit based on the information in this dict. LLNL/units uses a complicated heuristic for this purpose.

Let's consider two examples:

1.
```python
>>> v1 = sc.scalar(1.0, unit='Å')
>>> v1.unit.to_dict()
{'__version__': 2, 'multiplier': 1e-10, 'powers': {'m': 1}}
>>> v1
<scipp.Variable> ()    float64             [Å]  1
```
Here, the units library recognises that `1e-10 m` equals `Å` and so uses that as the output. Do you agree that we want to keep this behaviour?

2.
```python
>>> v2 = sc.scalar(1.0, unit='m') * sc.scalar(1.0, unit='dm')
>>> v2.unit.to_dict()
{'__version__': 2, 'multiplier': 0.1, 'powers': {'m': 2}}
>>> v2
<scipp.Variable> ()    float64         [0.1m^2]  1
```
Here, the library does not know a better name for `0.1 m^2` and so outputs the unit with a multiplier. This is the case you would like to change, correct?

If we wanted to preserve case 1 and change case 2 to output `<scipp.Variable> ()    float64         [m^2]  0.1`, we would have to build our own heuristic on top of the one in LLNL/units to figure out when the outputted unit is undesirable and then apply the scale factor to the values.

**Comment by SimonHeybrock** [comment]
I think it gets even more complicated: In operations with more than two operands such a logic would result in "unpredictable" results and multiple conversions. For example, `mm * m / mm` would convert twice, even though it is not necessary, and operations would become non-associative.

**Comment by damskii9992** [comment]
They would only become non-associative in regards to the displayed unit, assuming no rounding error in the conversions. And while I do see the issue with multiple conversions for performance reasons, I wonder if this will really have enough of an impact on performance to warrant not doing it, in a real-life use-case scenario?

**Comment by SimonHeybrock** [comment]
```
(mm * m) / m  # result in meter
mm * (m / m)  # result in millimeter
```
or am I missing something?

**Comment by damskii9992** [comment]
No, that is exactly what I am referring to, the results are numerically the same (except for rounding errors), but the units are different. So they are only non-associative in regards to the final displayed unit.

**Comment by nvaytet** [comment]
I think we should have an in-person meeting about this. This discussion is going on for too long.

**Comment by SimonHeybrock** [comment]
For reference: Scipp's behavior of not moving scale factors into the values automatically is consistent with Pint and AstroPy and I think this is desirable and important (and avoids a series of other problems, as touched on above):
<img width="354" alt="image" src="https://github.com/user-attachments/assets/4d6fd21d-e40c-4bdb-a725-d750b98e337c">

I believe any discussion should therefore center on improving the formatting.

**Comment by SimonHeybrock** [comment]
Conclusion from group discussion:
- Do not change behavior of `*` or `/` operators.
- Consider contributing or suggesting improved formatting upstream, even though this does not work in dimensionless cases (such as `m/km`).
